packages/analytics-plugin-google-global-site-tag/src/browser.ts

```diff
@@ -0,0 +1,528 @@
+ * @param {String} [campaignData.campaignMedium] - Medium of campaign
+ * @param {String} [campaignData.campaignContent] - Content of campaign
+ * @param {String} [campaignData.campaignKeyword] - Keyword of campaign
+ */
+function addCampaignData(campaignData: ICampaignDataProps = {}) {
+  let campaign: ICampaign = {};
+  const { name, source, medium, content, keyword } = campaignData;
+  if (name) campaign.campaignName = name;
+  if (source) campaign.campaignSource = source;
+  if (medium) campaign.campaignMedium = medium;
+  if (content) campaign.campaignContent = content;
+  if (keyword) campaign.campaignKeyword = keyword;
+  return campaign;
+}
+
+/**
+ * Changes format of custom dimensions from:
+ * { traitOne: 'dimension1', traitTwo: 'dimension2' }
+ * to:
+ * { dimension1: traitOne, dimension2: traitTwo }
+ */
+const formatCustomDimensionsIntoCustomMap = (plugin: IPluginConfig) => {
+  const { customDimensions } = plugin;
+  return customDimensions && Object.entries(customDimensions as { [key: string]: string }).reduce((acc: { [key: string]: string }, entry: [string, string]) => {
+    const [key, value] = entry;
+    acc[value] = key;
+    return acc;
+  }, {});
+}
+
+/**
+ * Create parameter value object based on given key values.
+ * */
+const getProperties = (properties: { [key: string]: unknown } = {}, propertyKeys: string[] = []) => {
+  return propertyKeys.reduce((acc: { [key: string]: unknown }, key: string) => {
+    if (properties[key]) {
+      acc[key] = properties[key];
+    }
+    return acc;
+  }, {});
+}
+
+/**
+ * Create custom dimension value object with keys that exist in
+ * config.customDimensions. If `config.setCustomDimensionsToPage`
+ * is true, set custom dimension values with the `set` command.
+ * Otherwise, return object for `config`.
+ * */
+const setCustomDimensions = (properties: { [key: string]: unknown } = {}, config: IPluginConfig) => {
+  const { customDimensions } = config;
+  if (!customDimensions) return {};
+
+  const customDimensionsValue = Object.keys(customDimensions as { [key: string]: string }).reduce((acc: { [key: string]: unknown }, key: string) => {
+    let value = get(properties, key) || properties[key];
+    if (typeof value === 'boolean') {
+      value = value.toString()
+    }
+    if (value || value === 0) {
+      acc[key] = value
+      return acc
+    }
+    return acc
+  }, {});
+
+  if (!Object.keys(customDimensionsValue).length) return {};
+
+  if (!config.setCustomDimensionsToPage) {
+    return customDimensionsValue;
+  }
+
+  window.gtag('set', customDimensionsValue);
+  return {};
+}
+
+export const identifyVisitor = (
+  id: string | undefined,
+  traits: { [key: string]: unknown } = {},
+  config: IPluginConfig = {},
+) => {
+  const trackingId = config.trackingId;
+  if (!window.gtag || !trackingId) return;
+  
+  if (id) {
+    window.gtag('set', { user_id: id });
+  }
+
+  if (Object.keys(traits).length) {
+    window.gtag('set', traits);
```

Comment:
With `gtag.js`, we no longer need to format the trait object into the form `{ [key: dimension[0-9]+]: value }`. Traits here are passed straight to `gtag`